### PR TITLE
Implemented attributes

### PR DESCRIPTION
**Pull request by PEMapModder**
- [x] basic attributes system
- [x] hunger mechanism
- [ ] experience mechanism
- [x] effects
  - [x] night vision
  - [x] blindness
  - [x] hunger
    - [x] declaration
    - [x] implementation
- [ ] exhaustion
  - [x] attack
  - [x] damage
  - [x] break blocks
  - [ ] walk per meter: 0.01
  - [ ] sneak per meter: 0.005
  - [ ] swim per meter: 0.015
  - [ ] sprint per meter: 0.1
  - [ ] jump: 0.2
  - [x] regen per halfheart | food >= 18: 4.0
- [x] events
  - [x] exhaust event
  - [x] eat event
- [x] data saving

Resolves:
* #3922
* #3830 
* #3809
* #3412
* Possible:
  * #3894

### CODE REVIEW

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -247,7 +242,8 @@ public function getLeaveMessage(){
 
 	/**
 	 * This might disappear in the future.
-	 * Please use getUniqueId() instead (IP + clientId + name combo, in the future it'll change to real UUID for online auth)
+	 * Please use getUniqueId() instead (IP + clientId + name combo, in the future it'll change to real UUID for online
+	 * auth)
```
Comment (robske110) [inline src/pocketmine/Player.php]: Thats messed up

Comment (PEMapModder) [inline src/pocketmine/Player.php]: PhpStorm auto reformat. Isn't a big issue.

**Comment by iksaku** [comment]
Hmm... You made it before I pushed all my local work :$

**Comment by PocketKiller** [comment]
@PEMapModder events should be ticked.

**Comment by robske110** [comment]
It isn't an event if it isn't ticked from somewhere ;)
[The event needs to be called and pocketmine is based on an tick loop]

**Pull request closed without merging**